modules/oe_migration_views/src/Plugin/views/field/MigrateMessagesField.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_migration_views\Plugin\views\field;
+
+use Drupal\views\Plugin\views\field\FieldPluginBase;
+use Drupal\views\ResultRow;
+
+/**
+ * Defines a field handler to display all migrate messages.
+ *
+ * @ingroup views_field_handlers
+ *
+ * @ViewsField("oe_migration_views_migrate_messages")
+ */
+class MigrateMessagesField extends FieldPluginBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function render(ResultRow $values) {
+    if (!isset($this->configuration['message_table'])) {
+      // @TODO Log/print or throw exception?
+      return '';
+    }
+
+    $source_ids_hash = $this->getValue($values);
+    $result = $this->getMigrateMessages($this->configuration['message_table'], $source_ids_hash);
+    $levels = _oe_migration_views_migrate_message_level_options();
+
+    $rows = [];
+    foreach ($result as $message_row) {
+      $level = isset($levels[$message_row->level]) ? $levels[$message_row->level] : $this->t('Unknown');
+      $rows[] = strtoupper($level) . ': ' . $message_row->message;
```

Comment:
`$level` is an object and when it's passed as parameter to `strtoupper()` it throws a `TypeError` exception. This only happens when there are messages to display.
Probably what you meant to do was `strtoupper($level->getUntranslatedString())`